django_sorcery/db/query.py

```diff
@@ -33,6 +70,61 @@ def get(self, *args, **kwargs):
 
         return super(Query, self).get(*args, **kwargs)
 
+    def filter(self, *args, **kwargs):
+        """
+        Standard SQLAlchemy filtering plus django-like expressions can be provided:
+
+        For example::
+
+            MyModel.objects.filter(MyModel.id == 5)
+            MyModel.objects.filter(id=5)
+            MyModel.objects.filter(id__gte=5)
+            MyModel.objects.filter(relation__id__gte=5)
+        """
+        args = args + tuple(self._lookup_to_expression(k, v) for k, v in kwargs.items())
+        return super(Query, self).filter(*args)
+
+    def _lookup_to_expression(self, lookup, value):
+        parts = lookup.split(LOOKUP_SEP)
+        info = meta.model_info(self._only_full_mapper_zero("get"))
+
+        props = dict(info.column_properties)
+        lhs = None
+
+        for i, part in enumerate(parts, 1):
+            is_last = i == len(parts)
+
+            if part in LOOKUP_TO_EXPRESSION:
+                return LOOKUP_TO_EXPRESSION[part](lhs, value)
+
+            elif part in info.relationships:
+                rel = info.relationships[part]
+                pairs = rel.local_remote_pairs_for_identity_key
+
+                # directly comparing to model
+                # e.g. .filter(relation=instance)
+                # which means we should filter by FK to avoid additional joins
```

Comment:
sqla does this already afaik, need to check

Reply:
hmm. actually there is another edge case which might need to be addressed in future pr as its more complicated to handle correctly - `.filter(relation__in=[Foo(id=1), Foo(id=2)])`, in django orm it will normalize down to `.filter(relation_id__in=[1,2])`. this is a bit more complex to handle as its not an obvious edge case in code but instead `in` lookup must be handling that directly... to throw my curve balls, thats cant break querysets - `.filter(relation__id=Foo.objects.filter(...))` so that check has to be specifically on lists or tuples or some other concrete iterable (non-generator)....